crypto/asn1/asn1_lib.c

```diff
@@ -275,18 +275,24 @@ ASN1_STRING *ASN1_STRING_dup(const ASN1_STRING *str)
     return ret;
 }
 
-int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len)
+int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len_in)
 {
     unsigned char *c;
     const char *data = _data;
+    size_t len;
 
-    if (len < 0) {
+    if (len_in < 0) {
         if (data == NULL)
             return 0;
-        else
-            len = strlen(data);
+        len = strlen(data);
+    } else {
+        len = (size_t)len_in;
+    }
+    if (len > SIZE_MAX - 1) {
```

Comment:
This should be INT_MAX-1 actually.

Reply:
Why exactly?  The `memcpy(2)` allows a `size_t` and passing a negative length value means calculate the length and hasn't limited it in the past.

BTW: I'm fine reducing this to `INT_MAX - 1` if it is desirable.

Reply:
Because then on line 304 you're assigning the value to int value str->length.

Reply:
Good point, I'll fix.

Reply:
Fixed now :)